framework/include/mesh/MooseMesh.h

```diff
@@ -1891,6 +1897,20 @@ MooseMesh::coordTransform()
   return *_coord_transform;
 }
 
+inline bool
+MooseMesh::hasCoordTransform() const
+{
+  if (parameters().isParamValid("alpha_rotation"))
+    return true;
+  if (parameters().isParamValid("beta_rotation"))
+    return true;
+  if (parameters().isParamValid("gamma_rotation"))
+    return true;
+  if (parameters().isParamSetByUser("up_direction"))
+    return true;
+  return false;
+}
```

Comment:
move this logic to MooseAppCoordTransform and use attributes instead of parameters